OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -111,15 +113,17 @@ public override bool HandleMouseInput(MouseInput mi)
 
 						if (unit != null && unit.Owner == (World.RenderPlayer ?? World.LocalPlayer))
 						{
-							var newSelection2 = SelectActorsInBox(World, worldRenderer.Viewport.TopLeft, worldRenderer.Viewport.BottomRight,
-								a => a.Owner == unit.Owner && a.Info.Name == unit.Info.Name);
+							// Select actors on the screen that have the same selection class as the already selected actor
+							var sc = unit.Trait<Selectable>().Class;
```

Comment:
See if ScreenMap.WithHighestSelectionPriority() doesn't guarantee that
already.
On Jun 8, 2015 4:12 PM, "Matija Hustić" <notifications@github.com> wrote:

> In OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs
> <https://github.com/OpenRA/OpenRA/pull/8212#discussion_r31911971>:
>
> > @@ -111,15 +113,17 @@ public override bool HandleMouseInput(MouseInput mi)
> >
> >  						if (unit != null && unit.Owner == (World.RenderPlayer ?? World.LocalPlayer))
> >  						{
> > -							var newSelection2 = SelectActorsInBox(World, worldRenderer.Viewport.TopLeft, worldRenderer.Viewport.BottomRight,
> > -								a => a.Owner == unit.Owner && a.Info.Name == unit.Info.Name);
> > +							// Select actors on the screen that have the same selection class as the already selected actor
> > +							var sc = unit.Trait<Selectable>().Class;
>
> Nope. Nice bug spotting there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenRA/OpenRA/pull/8212/files#r31911971>.
>